### ISSUE TITLE
A way to render synchronously

### ISSUE THREAD

**Issue by dmail** [preactjs/preact#4044]
**Describe the feature you'd love to see**

I need a way to update the UI synchronously using preact. Ideally something like [flushSync](https://react.dev/reference/react-dom/flushSync).

**Additional context**

In my use case: 
- preact is rendering some UI
- code outside preact is updating the surrounding of that UI 
- when it happens we want to hide a button in the UI rendered by preact

Currently we see an intermediate state where the surrounding is updated and button is still displayed. 
This intermediate state lasts <1ms, the time for preact to update the UI.
Intermediate states are undesirable as they cause blinks on the button. 

With react I could do something like that:

```js
import { flushSync } from "react-dom";
import { useState } from "react";

window.setButtonVisibility = () => {}; // called by external code

export const App = () => {
  const [visible, setVisible] = useState();
  window.setButtonVisibility = (value) => {
    flushSync(() => {
      setVisible(value);
    });
  };
  return <button style={{ display: visible ? "block": "none" }}>Hello</button>;
}
```

But in preact I don't know what I could do

**Related**

- https://github.com/preactjs/preact/issues/3929
- https://github.com/preactjs/preact/issues/3925

**Comment by JoviDeCroock** [comment]
You can leverage something like

```
import { options } from "preact";

function myCrucialUpdater() {
  let oldDebounce = options.debounceRendering;
  options.debounceRendering = fn => fn();
  update();
  options.debounceRendering = oldDebounce;
}
```

**Comment by dmail** [comment]
Ohhh, nice 💯

**Comment by JoviDeCroock** [comment]
Hey @dmail did this work out for you?

**Comment by dmail** [comment]
Hi @JoviDeCroock, 

In the end we found a second source of desynchronization. So rendering synchronously was not enough to fix our issue. 

In our scenario we have a root page without preact embedding an iframe with UI rendered by preact.
The communication between the parent page and the iframe is done via postMessage.
When the parent page tells to the iframe to update the internal state via postmessage the browser waits a bit before executing js listening inside the iframe. So preact is notified too late, rendering sycnrhonously would not be enough to catch up.
In our case we decided to let parent page wait for the iframe to react to the postmessage before updating UI in the parent page. This way there is no intermediate state displayed to the user.
